### PR TITLE
Implement HeatingActuator

### LINKED ISSUE #76: Add HeatingActuator as valve entity

### PR DESCRIPTION
**Pull request by derjoerg**
closes #76

### CODE REVIEW

File: custom_components/abbfreeathome_ci/valve.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    @property
+    def unique_id(self) -> str | None:
+        """Return a unique ID."""
+        return f"{self._valve.device_id}_{self._valve.channel_id}_valve"
+
+    @property
+    def supported_features(self) -> int | None:
+        """Return supported features."""
+        return ValveEntityFeature.SET_POSITION | ValveEntityFeature.CLOSE
```
[the remark this answers is not in the document]
Comment (derjoerg) [reply custom_components/abbfreeathome_ci/valve.py:87]: I understood the sentence in a way that only the `method` doesn't needs to be implemented.
`CLOSE` can be easily achieved as HA only needs to call the `set_position` function with 0.

I also first only implemented `SET_POSITION`, but `CLOSE` just added the possibility to switch the binary switch to off.

But to be honest, I don't plan to use these devices in my setup (that's why I keep them disabled by default), so if you want I change it :smiley:

[the remark this answers is not in the document]
Comment (kingsleyadam) [reply custom_components/abbfreeathome_ci/valve.py:87]: As long as it works, I'm ok with it. The class really only support set position so adding the close feature seemed odd.

File: custom_components/abbfreeathome_ci/valve.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        super().__init__()
+        self._valve = valve
+        self._sysap_serial_number = sysap_serial_number
+
+        self.entity_description = ValveEntityDescription(
+            key="HeatingActuatorValve",
+            device_class=ValveDeviceClass.WATER,
+            name=valve.channel_name,
+            reports_position=True,
+        )
```
Comment (kingsleyadam) [inline custom_components/abbfreeathome_ci/valve.py:52]: Sorry, I missed this the first time. Anytime I can specify a property in the entity_description I do, this ensure's we don't always overwrite the Home Assistant property code and allows for the HomeAssistant to apply it's own logic.

```suggestion
        self.entity_description = ValveEntityDescription(
            key="HeatingActuatorValve",
            device_class=ValveDeviceClass.WATER,
            entity_registry_enabled_default=False,
            name=valve.channel_name,
            reports_position=True,
        )
```

File: custom_components/abbfreeathome_ci/valve.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    @property
+    def supported_features(self) -> int | None:
+        """Return supported features."""
+        return ValveEntityFeature.SET_POSITION
+
+    @property
+    def entity_registry_enabled_default(self) -> bool | None:
+        """Return if the device should be enabled by default."""
+        return False
```
Comment (kingsleyadam) [inline custom_components/abbfreeathome_ci/valve.py:92]: See above.

```suggestion
```

**Review by kingsleyadam — CHANGES_REQUESTED** [review]
Last comment, do we want to provide a translation for this entity?

Looking at the example device info: https://github.com/kingsleyadam/local-abbfreeathome/issues/75

The heating actuator name is the same as the device name, usually we're specifying a translated text in this scenario e.g. `Heating Valve`.

**Comment by derjoerg** [comment]
> Last comment, do we want to provide a translation for this entity?
> 
> Looking at the example device info: [kingsleyadam/local-abbfreeathome#75](https://github.com/kingsleyadam/local-abbfreeathome/issues/75)
> 
> The heating actuator name is the same as the device name, usually we're specifying a translated text in this scenario e.g. `Heating Valve`.

At least in my setup each valve has its own name

**Pull request merged**